src/app/store/machine/utils/hooks.ts

```diff
@@ -20,6 +21,24 @@ import {
   isNodeStorageConfigurable,
 } from "app/store/utils";
 import vlanSelectors from "app/store/vlan/selectors";
+import { isId } from "app/utils";
+
+/**
+ * Get a machine via the API.
+ * @param id - A machine's system id.
+ */
+export const useGetMachine = (id?: Machine[MachineMeta.PK] | null): void => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    // TODO: this should not fetch the machine again once the request has been
+    // made. This can be done by checking the request id once the action has
```

Comment:
I think we should try to avoid storing a lot of information on the front end to limit complexity (if anything only store a single machine at a time).

Reply:
Could you explain a bit more about what you mean? The UI needs to be able to fetch and display multiple machines at once. The machine will be removed once it no longer needs to be displayed anywhere the UI. This comment is about not fetching the same machine multiple times from the same component.